azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,494 @@
+                resource_group_name=resource_group,
+                azure_key_vault_secret_provider_class_name=OPCUA_SPC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_spc = {}
+
+        # check if there is a secret sync called "aio-opc-ua-broker-client-certificate", if not create one
+        try:
+            opcua_secret_sync = self.ssc_mgmt_client.secret_syncs.get(
+                resource_group_name=resource_group,
+                secret_sync_name=OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_secret_sync = {}
+
+        secrets_to_add = []
+        for file in [public_key_file, private_key_file]:
+            file_name = os.path.basename(file)
+            file_name_info = os.path.splitext(file_name)
+            cert_extension = file_name_info[1].replace(".", "")
+            cert_name = file_name_info[0].replace(".", "-")
+            secret_name = f"{cert_name}-{cert_extension}"
+
+            # iterate over secrets to check if secret with same name exists
+            secret_name = self._check_and_update_secret_name(secrets, secret_name, spc_keyvault_name)
+            self._upload_to_key_vault(secret_name, file, cert_extension)
+            secrets_to_add.append((secret_name, file_name))
+
+        self._add_secrets_to_spc(
+            secrets=[secret[0] for secret in secrets_to_add],
+            spc=opcua_spc,
+            resource_group=resource_group,
+            spc_keyvault_name=spc_keyvault_name,
+            spc_tenant_id=spc_tenant_id,
+            spc_client_id=spc_client_id,
+        )
+
+        self._add_secrets_to_secret_sync(
+            secrets=secrets_to_add,
+            secret_sync=opcua_secret_sync,
+            resource_group=resource_group,
+            spc_name=OPCUA_SPC_NAME,
+            secret_sync_name=OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
+        )
+
+        # update opcua extension
+        return self._update_client_secret_to_extension(
+            subject_name=subject_name,
+            application_uri=application_uri,
+        )
+
+    def _validate_key_files(self, public_key_file: str, private_key_file: str):
+        # validate public key file end with .der
+        validate_file_extension(public_key_file, [".der"])
+        # validate private key file end with .pem
+        validate_file_extension(private_key_file, [".pem"])
+
+        # validate public key and private key has matching file name without extension
+        public_key_name = os.path.basename(public_key_file)
+        public_key_name = os.path.splitext(public_key_name)[0]
+        private_key_name = os.path.basename(private_key_file)
+        private_key_name = os.path.splitext(private_key_name)[0]
+
+        if public_key_name != private_key_name:
+            raise ValueError(f"Public key file {public_key_name} and private key file {private_key_name} must match.")
```

Comment:
flopping on if the public key file + private key file should just be one (assuming it is required to be the same)... but if it is two then it will better imply that two files are needed

Reply:
both public key file and private key file are required at the cmd level, this is just to check the public key file and private key file name has to match exactly